src/miner.cpp

```diff
@@ -419,6 +450,12 @@ void BlockAssembler::addPackageTxs(int &nPackagesSelected, int &nDescendantsUpda
         // This transaction will make it in; reset the failed counter.
         nConsecutiveFailed = 0;
 
+        if (min_package_fee_rate) {
+            // Compare package fee rate and potentially update new minimum
+            CFeeRate newFeeRate(packageFees, packageSize);
+            if (newFeeRate < *min_package_fee_rate) *min_package_fee_rate = newFeeRate;
```

Comment:
Earlier in this function, there's code to skip transactions if they're too large for the block. I think this can result in the mempool containing txs with fee rates of: `120 s/vb, 119 s/vb, 118 s/vb, 117 s/vb, 115 s/vb, 110 s/vb, 105 s/vb, 100 s/vb, 95 s/vb, 80 s/vb, 79 s/vb, ...` and the block containing txs with fee rates of: `120, 119, 118, 117, 95, 80` -- because the 110-100 s/vb txs just didn't fit in the block, but the 95 and 80 s/vb txs happened to be tiny. This would then incorrectly result in txs with fee rates above 80 s/vb being rebroadcast, whereas really only txs with fee rates above 117 s/vb should be.